endpoints/users.py

```diff
@@ -1,71 +1,210 @@
-from flask import Flask, jsonify
-
-from example_pb2 import IncreaseUserCount, ExampleResponse
-
-from kafka import KafkaProducer, KafkaConsumer
-from kafka.errors import NoBrokersAvailable
-
-import threading
-import time
-
-app = Flask(__name__)
-
-KAFKA_BROKER = "kafka-broker:9092"
-brokers_available = False
-producer = None
-
-while not brokers_available:
-    try:
-        producer = KafkaProducer(bootstrap_servers=[KAFKA_BROKER])
-        consumer = KafkaConsumer(
-                'greetings', 
-                bootstrap_servers = [KAFKA_BROKER], 
-                auto_offset_reset='earliest',
-                group_id = 'api-endpoint-users'
-            )
-        brokers_available = True
-        print("Got the broker!", flush=True)
-    except NoBrokersAvailable:
-        time.sleep(4)
-        continue
-
-@app.route('/user/<string:name>/add')
-def increment_user(name):
-    """ Sends an increment request to the specified name"""
-    global producer, consumer
-
-    print("Received request to increment name ",name)
-    request = IncreaseUserCount()
-    request.name = name
-
-    if producer is None:
-        print("Creating the producer", flush=True)
-        producer = KafkaProducer(bootstrap_servers=[KAFKA_BROKER])
-        print("Got the broker!", flush=True)
-
-
-    key = name.encode('utf-8')
-    val = request.SerializeToString()
-    producer.send(topic="names", key=key, value=val)
-    producer.flush()
-
-    # try to get the message from the consumer
-    for msg in consumer:
-        print(msg, flush=True)
-        print(msg.key.decode('utf-8'), flush=True)
-        if msg.key.decode('utf-8') == name:
-            print("Got a response for the user {} --> {}".format(name, msg.value), flush=True)
-
-            return jsonify({'response': msg.value.decode('utf-8')})
-
-@app.route('/test/<string:name>')
-def test(name):
-    return f"It's working {name}!!"
-
-@app.route('/')
-def basic():
-    return "Hello world!"
-    
-
-if __name__ == "__main__":
-    app.run()
+from flask import Flask, Response
+
+# Import the messages to be sent to the statefun cluster
+from users_pb2 import CreateUserRequest
+from users_pb2 import UserRequest, UserResponse
+
+from kafka import KafkaProducer, KafkaConsumer
+from kafka.errors import NoBrokersAvailable
+
+import threading
+import time
+import uuid
+import json
+
+from typing import Dict
+
+# create the logger and configure
+import logging
+
+FORMAT = '[%(asctime)s] %(levelname)-8s %(message)s'
+logging.basicConfig(level=logging.INFO, format=FORMAT)
+
+logger = logging.getLogger()
+
+
+
+
+# Some parameters to send and read from kafka
+KAFKA_BROKER = "kafka-broker:9092"

```

Comment:
Maybe we should configure this using a command line argument. We can also do this later ofc.